bsmetadata/metadata_utils.py

```diff
@@ -183,14 +183,33 @@ def _collate_metadata(metadata_list: List[dict], cfg: MetadataConfig):
         assert metadata_node.relative_start_pos not in metadata_dict_idx[metadata_node.char_start_idx]
         assert metadata_node.relative_end_pos not in metadata_dict_idx[metadata_node.char_end_idx]
 
-        metadata_dict_idx[metadata_node.char_start_idx][metadata_node.relative_start_pos] = start_text
-        metadata_dict_idx[metadata_node.char_end_idx][metadata_node.relative_end_pos] = end_text
+        if start_text:
+            metadata_dict_idx[metadata_node.char_start_idx][metadata_node.relative_start_pos] = start_text
+        if end_text:
+            metadata_dict_idx[metadata_node.char_end_idx][metadata_node.relative_end_pos] = end_text
 
     for absolute_idx, value in metadata_dict_idx.items():
         pos_sorted = sorted(list(value.keys()))
         local_metadata = ""
         for pos in pos_sorted:
             local_metadata += metadata_dict_idx[absolute_idx][pos]
+
+        # We add here a local special token if needed around the metadata list of a type if needed
+        if (
+            cfg.local_metadata_special_token_start
+            and local_metadata
+            and metadata_list[0]["key"] in cfg.local_metadata_special_token_start
+        ):
+            local_special_token_start = cfg.local_metadata_special_token_start[metadata_list[0]["key"]]
+            local_metadata = f"{local_special_token_start}{local_metadata}"
+        if (
+            cfg.local_metadata_special_token_end
+            and local_metadata
+            and metadata_list[0]["key"] in cfg.local_metadata_special_token_end
+        ):
+            local_special_token_end = cfg.local_metadata_special_token_end[metadata_list[0]["key"]]
+            local_metadata = f"{local_metadata}{local_special_token_end}"
+
```

Comment:
Two minor questions:

1. Just curious, is it possible that a `metadata_list[0]["key"]` can be in both `cfg.local_metadata_special_token_start` and  `cfg.local_metadata_special_token_end`? I understand that we usually have an opening-closing tag pair for those special tokens, so technically it should never happen. If that is always the case, then perhaps we can have one `cfg.local_metadata_special_token_something` and then control the opening/closing by `cfg,entity_setting`, if I read the code correctly.
2. I guess we don't have to deal with nested `ENTITY_CHAIN`s because impossible by design?

---
> ℹ️ Mike's reviewing oath 🖖
> 1. I will abstain from all intentional wrong-doing and harm.
> 2. I will refrain myself from submitting nitpicking software-engineering-related comments.
> 3. I will take the time to clarify and justify my suggestions.